test/dashboard-e2e/helpers/common.ts

```diff
@@ -12,13 +13,25 @@ export function validateTest(testData: Partial<TestData>, createdTestData: TestD
   expect(testData.name).toEqual(createdTestData.name);
   validateLabels(testData.labels, createdTestData.labels);
   expect(testData.type).toEqual(createdTestData.type);
-  expect(testData.content.type).toEqual(createdTestData.content.type);
 
   // testSources
   if (testData.content.type === 'git') {
+    expect(testData.content.type).toEqual(createdTestData.content.type);
+
     Object.keys(testData.content.repository).forEach(key => {
       expect(testData.content.repository[key]).toEqual(createdTestData.content.repository[key]);
     });
+  } else if (testData.content.type === 'string') {
+    expect(testData.content.type).toEqual(createdTestData.content.type);
+
+    Object.keys(testData.content).forEach(key => {
+      expect(testData.content[key]).toEqual(createdTestData.content[key]);
+    });
```

Comment:
```suggestion
     expect(createdTestData.content).toMatchObject(testData.content);
```

Reply:
I iterate over `testData.content` keys on purpose here - response still contain redundant things (for example, empty repository) even if a test is created with string type

Reply:
`expect(obj1).toMatchObject(obj2)` is checking if all of the properties in `obj2` are matching in the `obj1`, so all not recognized properties will be ignored.